### PR TITLE
[PROCS-4045] Move process agent statsd client initialization into process agent component

### PR DESCRIPTION
**Pull request by daniel-taf**
### What does this PR do?

This PR moves process agent statsd client initialization into the process agent component constructor. 

### Motivation

This helps to ensure the statsd client is configured before the checks are run and so that the statsd client can be used when the checks are running on the core agent. Otherwise, the client would be noop and no metrics could be reported from the checks.

### Additional Notes

While this does result in redundant network calls, the alternative for maintaining the same functionality is to convert the process checks into a collector check which is not easily done at the moment.

### Possible Drawbacks / Trade-offs

Redundant network calls on the core agent.

### Describe how to test/QA your changes

Run the process checks on the core agent and verify metrics (`datadog.process.processes.host_count`) are reported with correct tags.
```
process_config:
  process_collection:
    enabled: true
  run_in_core_agent:
    enabled: true
```

### CODE REVIEW

File: releasenotes/notes/process-statsd-client-fix-66b11cc991a9f6be.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+fixes:
+  - |
+    Fix metric reporting for process checks on the core agent.
```
Comment (brett0000FF) [inline releasenotes/notes/process-statsd-client-fix-66b11cc991a9f6be.yaml:11]:
```suggestion
    Fix metric reporting for process checks on the core Agent.
```

**Review by brett0000FF — APPROVED** [review]

File: comp/process/agent/agentimpl/agent.go [hunk 2]
```diff
@@ -96,6 +100,15 @@ func newProcessAgent(deps dependencies) provides {
 		}
 	}
 
+	if err := processStatsd.Configure(ddconfig.GetBindHost(), deps.Config.GetInt("dogstatsd_port"), deps.Statsd.CreateForHostPort); err != nil {
+		deps.Log.Criticalf("Error configuring statsd for process-agent: %s", err)
```
[the remark this answers is not in the document]
Comment (daniel-taf) [reply comp/process/agent/agentimpl/agent.go:104]: Makes sense. Done

File: comp/process/agent/agentimpl/agent.go [hunk 3]
```diff
@@ -14,13 +14,16 @@ import (
 	logComponent "github.com/DataDog/datadog-agent/comp/core/log"
 	statusComponent "github.com/DataDog/datadog-agent/comp/core/status"
 	"github.com/DataDog/datadog-agent/comp/core/sysprobeconfig"
+	compstatsd "github.com/DataDog/datadog-agent/comp/dogstatsd/statsd"
```
[the remark this answers is not in the document]
Comment (daniel-taf) [reply comp/process/agent/agentimpl/agent.go]: Done

**Review by dustmop — APPROVED** [review]
LGTM

**Review by robertjli — APPROVED** [review]

**Pull request merged**